### PR TITLE
[GSoC] Function for I.L.D. for Shear added

### PR DESCRIPTION
**Pull request by Psycho-Pirate**
#### References to other Issues or PRs

See #19069

#### Brief description of what is fixed or changed
Function `solve_for_ild_shear()` is added which calculates the shear equation under the effect of a moving load.
Function `plot_ild_shear()` is added which plots these equations.

#### Release Notes

* physics.continuum_mechanics 
   * Added ability to calculate and plot I.L.D. Shear Equations

### CODE REVIEW

**Comment by Psycho-Pirate** [comment]
@ishanaj

File: sympy/physics/continuum_mechanics/beam.py [hunk 1]
```diff
@@ -1728,6 +1741,147 @@ def plot_ild_reactions(self, subs=None):
...
+
+        x = self.variable
+        l = self.length
+
+        shear_force = self._solve_for_ild_shear()
+
+        shear_curve1 = val - limit(shear_force, x, dist)
+        shear_curve2= (limit(shear_force, x, l)-limit(shear_force, x, dist))-val
+
+        for i in reactions:
```
Comment (ishanaj) [inline sympy/physics/continuum_mechanics/beam.py:1810]: Avoid using a vague variable name like `i`, should rather be `reaction`

File: sympy/physics/continuum_mechanics/beam.py [hunk 2]
```diff
@@ -1728,6 +1741,147 @@ def plot_ild_reactions(self, subs=None):
...
+        Helper function for solve_for_ild_shear(). It takes the unsubstituted
+        copy of the load equation and uses it to calculate shear force equation.
+
+        """
+        x = self.variable
+        shear_force = -integrate(self._original_load, x)
+
+        return shear_force
+
+    def solve_for_ild_shear(self, dist, val, *reactions):
```
Comment (ishanaj) [inline sympy/physics/continuum_mechanics/beam.py:1756]: It should rather be `distance` and `value`. This change should then also be done for `solve_for_ild_reactions()`

File: sympy/physics/continuum_mechanics/beam.py [hunk 3]
```diff
@@ -1728,6 +1741,147 @@ def plot_ild_reactions(self, subs=None):
...
+        """
+
+        Determines the Influence Line Diagram equations for shear at a
+        specified point under the effect of a moving load.
+
+        Parameters
+        ==========
+        dist : Integer
+            Distance of the specified point
```
Comment (ishanaj) [inline sympy/physics/continuum_mechanics/beam.py:1765]: Elaborate this to specify that the point is where we intend to determine the shear and it is the distance from the start of the beam

**Comment by ishanaj** [comment]
We can discuss the possibility and ways to generate the shear eq in terms of `SingularityFunction`. But I think generating them in `Piecewise` won't be much of an issue.
Rest looks good to me!

**Comment by Psycho-Pirate** [comment]
I have made the changes @ishanaj

**Review by ishanaj — APPROVED** [review]

**Comment by ishanaj** [comment]
LGTM! Merging!

**Pull request merged**